validator/sawtooth_validator/journal/publisher.py

```diff
@@ -221,7 +222,7 @@ def on_check_publish_block(self, force=False):
                                                             _candidate_block):
                     candidate = self._candidate_block
                     self._candidate_block = None
-                    candidate = self._finalize_block(candidate)
+                    self._finalize_block(candidate)
```

Comment:
This call is fine ( I got it confused with the call above on line 199). This call sets the consensus data and generates the signature necessary before build block is called.